### PR TITLE
Treat URLs have same IP address as same

### PR DESCRIPTION
**Pull request by sorah**
- To solve validation error problem using URLs in hostname #2123

cc: @barakmich @kelseyhightower

### CODE REVIEW

**Comment by sorah** [comment]
This is a kind of short-term fix, so should be removed after some improvements...

**Comment by xiang90** [comment]
@sorah I have not explored into this much. But I think a better approach might be that we always resolves the dns to TCP addresses rather than delay it to the comparing face. So we will always have a "correct and consistent" representation of a hostname.

**Comment by barakmich** [comment]
LGTM for the short term, I think @xiang90  is right that the long term fix is to resolve any dns the minute we get it, and only deal in IPs.

File: etcdserver/cluster.go [hunk 1]
```diff
@@ -393,7 +393,7 @@ func ValidateClusterAndAssignIDs(local *Cluster, existing *Cluster) error {
 	sort.Sort(SortableMemberSliceByPeerURLs(lms))
 
 	for i := range ems {
-		if !reflect.DeepEqual(ems[i].PeerURLs, lms[i].PeerURLs) {
+		if !netutil.URLStringsEqual(ems[i].PeerURLs, lms[i].PeerURLs) {
```
Comment (xiang90) [inline etcdserver/cluster.go]: i would like to merge this pr if you add a todo here in case we forget this is a quick fix.

Comment (sorah) [inline etcdserver/cluster.go]: Annotated! f8ce599

**Comment by xiang90** [comment]
lgtm

**Comment by sorah** [comment]
@xiang90 Agreed about a better approach. but note that we have to keep both, for TLS hostname verification, as I noted in #2123.

**Pull request merged**